### ISSUE TITLE
Screen reader support for header week numbers breaks jest-axe tests

### ISSUE THREAD

**Issue by SheaClose** [Hacker0x01/react-datepicker#5807]
**Describe the bug**
The accessibility improvements added in https://github.com/Hacker0x01/react-datepicker/pull/5731 is triggering a failure in jest-axe when using `renderCustomHeader` due to missing role="row"

**To Reproduce**

1. Consume Datepicker component, providing a renderCustomHeader arg.
2. run jest-axe test against component.

**Expected behavior**
Should pass accessibilty testing when using renderCustomHeader

**Screenshots**

<img width="986" height="593" alt="Image" src="https://github.com/user-attachments/assets/6289e87c-260c-4088-9ff4-0379c36f90f3" />

**Desktop (please complete the following information):**

- OS: macOS Sonoma 14.6.1
- Browser n/a
- Version 8.5.0

**Comment by martijnrusschen** [comment]
Can you check out https://github.com/Hacker0x01/react-datepicker/pull/5808 to see if this fixes it?

**Comment by SheaClose** [comment]
This does appear to have resolved the issue!

**Comment by martijnrusschen** [comment]
I've merged the fix including additional jest-axe testing in this repo so we can hopefully catch these before we release them next time.
